tools/mkcomposefs.c

```diff
@@ -1690,8 +1690,12 @@ int main(int argc, char **argv)
 		buildflag_copy |= LCFS_BUILD_NO_INLINE;
 
 		root = lcfs_build(AT_FDCWD, src_path, buildflag_copy, &failed_path);
-		if (root == NULL)
-			err(EXIT_FAILURE, "error accessing %s", failed_path);
+		if (root == NULL) {
+			// An allocation error in maybe_join_path() can cause
+			// failed_path to be set to NULL
+			char *failed_path_not_null = failed_path ? failed_path : "";
```

Comment:
Two minor things:
- This can be `failed_path ?: ""`
- could also be `const char *` or just inline it at usage site (since `?` supports that)